adminSiteServer/apiRouter.ts

```diff
@@ -948,6 +948,36 @@ apiRouter.get("/variables.json", async (req) => {
     return { variables: rows, numTotalRows: numTotalRows }
 })
 
+apiRouter.get("/variables.usages.json", async (req) => {
+    // This is a bit of a workaround - what we would like is to select
+    // all variables and do a group by on the joined charts table with a count(*)
+    // but since the variables are stored inside the json config and mysql 5.7's limited
+    // support for json and arrays we can'd do it that way - so instead we select the arrays
+    // and create a map that we use to "transpose" the table, then serialize this
```

Comment:
There _shouldn't_ be a case where `chart_dimensions` is out of sync. If there is, we should probably fix it, because we depend on this foreign key reference to prevent data for charts being deleted.